### ISSUE TITLE
[New Resource]: Broadcast Domain

### ISSUE THREAD

**Issue by acch** [NetApp/terraform-provider-netapp-ontap#303]
### Description

Ability to create broadcast domains

### Requested Resource(s) and/or Data Source(s)

* netapp-ontap_broadcast_domain

### Potential Terraform Configuration

```terraform
// sample broadcast domain with required properties
resource "netapp-ontap_network_broadcast_domain" "example" {
  cx_profile_name = "cluster1"
  ipspace         = "Default"
  name            = "bd1"
  mtu             = "1500"
}
```

### References

* https://docs.netapp.com/us-en/ontap-restapi/ontap/network_ethernet_broadcast-domains_endpoint_overview.html
* https://docs.netapp.com/us-en/ontap-restapi/ontap/post-network-ethernet-broadcast-domains.html

### Would you like to implement a fix?

Yes

**Comment by acch** [comment]
Do you have any objections with the proposed changes? If not, then I will go ahead and start working on a PR to implement the new `broadcast_domain` data source and resource...

**Comment by carchi8py** [comment]
@acch yes go ahead and start working on this if you want. I'll go ahead and assign it to you.

For resources we try to have ACC test that run on a real system to test common error cases, (create/read/, update, and import). 

If you run with these variable
https://github.com/NetApp/terraform-provider-netapp-ontap/blob/integration/main/internal/provider/networking/network_ip_interface_resource_alias_test.go#L60C1-L62C45

The pipeline will run the ACC test on a Vsim we have set up.

If you need something set up on that Vsim to run the ACC tests let me know.

**Comment by acch** [comment]
I've prepared a PR to address this change: #333.

Please let me know if anything's missing.
